enterprise/server/sociartifactstore/sociartifactstore.go

```diff
@@ -319,15 +321,15 @@ func (s *SociArtifactStore) getArtifactsFromCache(ctx context.Context, imageId s
 		return nil, err
 	}
 	for _, ztocDigest := range ztocDigests {
-		exists, err := s.cache.Contains(ctx, resourceName(ztocDigest))
+		exists, err := s.cache.Contains(ctx, resourceName(ztocDigest, rspb.CacheType_CAS))
 		if err != nil {
 			return nil, err
 		}
 		if !exists {
 			return nil, status.NotFoundErrorf("ztoc %s not found in cache", ztocDigest.Hash)
 		}
 	}
-	return getArtifactsResponse(imageId, sociIndexDigest, ztocDigests), nil
+	return getArtifactsResponse(imageConfigHash, sociIndexDigest, ztocDigests), nil
 }
 
 // Serializes a repb.Digest as "<digest>/<size-bytes">
```

Comment:
We could replace `serializeDigest` and `deserializeDigest` with digest.ResourceName().DownloadString and digest.func ParseDownloadResourceName(resourceName string)

Reply:
Sounds good, done